README.md

```diff
@@ -51,66 +51,87 @@ npm install ramldt2jsonschema --save
-const includesPath = '/different/path/to/includes/'
-const ramlData = fs.readFileSync(filePath).toString()
+const filePath = join(__dirname, 'complex_cat.json')
+const jsonData = fs.readFileSync(filePath).toString()
 
-raml2json.dt2js.setBasePath(includesPath)
-const schema = raml2json.dt2js(ramlData, 'Cat')
-console.log(JSON.stringify(schema, null, 2))
+async function main () {
+  let raml
+  try {
+    raml = await r2j.js2dt(jsonData, 'Cat')
+  } catch (err) {
+    console.log(err)
+    return
+  }
+  console.log('#%RAML 1.0 Library\n')
+  console.log(yaml.safeDump(raml, { 'noRefs': true }))
+}
+
+main()
 ```
 
-#### js2dt
+#### Resolving references
+
+When the input contains external references (`!include`, `uses:`, `$ref`, etc.) and the referred files are not in the same directory as the script it is being ran from, you may provide a third argument `basePath` to both `dt2js` and `js2dt`. All references will then be resolved relative to that base path.
 
+Example of using `basePath` argument in dt2js:
 ```js
+// Script below ran from /home/john/where/ever/
+// Reference is located at /home/john/schemas/simple_person.json
 const raml2json = require('ramldt2jsonschema')
-const join = require('path').join
-const fs = require('fs')
 
-const filePath = join(__dirname, 'schema.json')
-const jsonData = fs.readFileSync(filePath).toString()
-
-const raml = raml2json.js2dt(jsonData, 'Person')
+const ramlStr = `
+  #%RAML 1.0 Library
 
-console.log(`#%RAML 1.0 Library\n${yaml.safeDump({ types: raml }, {'noRefs': true})}`)
+  types:
+    Person: !include simple_person.json
+`
+const basePath = '/home/john/schemas/' // or '../../schemas/'
+const schema = raml2json.dt2js(ramlStr, 'Person', basePath)
+console.log(JSON.stringify(schema, null, 2))
 ```
 
 ### Limitations
 
 - in js2dt,
-  - the following JSON schema properties are not supported and as a result, will not be converted:
-    - `not` (or `false` when used as a schema boolean)
-    - object's `dependencies`
-    - array's `additionalItems`
-    - the `propertyNames` keyword
-    - the `contains` keyword
-  - JSON schema number and integer's `exclusiveMaximum` and `exclusiveMinimum` properties are converted to raml 'maximum' and 'minimum'
-  - the JSON schema `oneOf` property is processed the same way as the JSON schema `anyOf` property, it is converted to the RAML `union` type
-  - the JSON schema `type: 'string'` with `media` and `binaryEncoding: 'binary'` is converted to the RAML `file` type
-  - the JSON schema `type: string` with a `pattern` that matches exactly one of the RAML date types will be converted in the matching RAML date type
-  - the JSON schema `media` property with `anyOf`, elements of which have the property `mediaType` will be converted to `fileTypes` in RAML
-
-- in dt2js,
-  - the following RAML properties are not supported/converted:
-    - annotations
+  - the following JSON Schema properties are not supported and as a result, will not be converted correctly:
```

Comment:
Updated the list and wording. We'll get a more complete list of limitations after #97 is implemented. It would take a long time to go through a list of all json schema properties and check how each of them is converted. Let me know if you want me to do that.

Reply:
then let's reword to `and as a result, may not be converted as expected` and add a task to `test & describe accurately the limitations in the README`

Reply:
Done. https://github.com/raml-org/ramldt2jsonschema/issues/100